server/schedule/operator_controller.go

```diff
@@ -166,21 +178,68 @@ func (oc *OperatorController) PushOperators() {
 // AddOperator adds operators to the running operators.
 func (oc *OperatorController) AddOperator(ops ...*Operator) bool {
 	oc.Lock()
-	defer oc.Unlock()
 
-	if oc.exceedStoreLimit(ops...) || !oc.checkAddOperator(ops...) {
+	if !oc.checkAddOperator(ops...) {
 		for _, op := range ops {
 			operatorCounter.WithLabelValues(op.Desc(), "canceled").Inc()
 			oc.opRecords.Put(op, pdpb.OperatorStatus_CANCEL)
 		}
+		oc.Unlock()
 		return false
 	}
-	for _, op := range ops {
-		oc.addOperatorLocked(op)
+
+	op := ops[0]
+	desc := op.Desc()
+	if oc.wopStatus[desc] == nil {
+		oc.wopStatus[desc] = &WaitingOperatorStatus{0, oc.cluster.GetSchedulerMaxWaitingOperator()}
+	}
+	if oc.wopStatus[desc].current >= oc.wopStatus[desc].max {
+		oc.Unlock()
+		return false
+	}
+	oc.wop.PutOperator(op)
+	if len(ops) > 1 {
```

Comment:
I know this is the merge-region case, however, I think it is better to add some comments here which is friendly to the community developers.